### DISCUSSION TITLE
Error loading ASGI app. Attribute "app" not found in module "main"

### DISCUSSION THREAD

**Discussion by maruf061** [fastapi/fastapi#4131] [Questions] [answered]
Hi Everyone!!

I am new to Python and FastAPI. 
I have installed the FastAPI and Python in my vscode and when I am running the command : uvicorn main:app , I am getting the above error message and localhost is not running.

Can anyone suggest me what should I do to resolve this problem?

**Comment by simondale00** [comment, marked as answer]
Hey @maruf061. The command `univorn main:app` expects to find a file named `main.py` in the current path. That file should then contain a variable named `app`, whose value is a valid FastAPI application.

**Reply by tiangolo** [reply]
Thanks @simondale00! :cake: 

@maruf061 if you still have problems, please create a minimal self-contained example that I can copy-paste to see the error.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by enparthasarathi** [comment]
File structure:
--------------------
       |-app
             |-main.py
             |- ..
             |- ..
      Dockerfile
      docker-compose.yml
.
.
.

Dockerfile:
------------
               CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8000", "--reload"]

INFO: Will watch for changes in these directories: ['/app']
INFO: Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO: Started reloader process [1] using StatReload
ERROR: Error loading ASGI app. Could not import module "main".

**Reply by simondale00** [reply]
I'd recommend checking the following:
* Is `app/main.py` in your Docker image?
* What is the working directory for the command being run? Is `main.py` within that path?

**Reply by enparthasarathi** [reply]
My docker file:
==========

WORKDIR /app

RUN pip install --no-cache-dir -r requirements.txt

COPY app/ .

CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8000", "--reload"]


Command argument in server:
---------------------------------
/usr/local/bin/uvicorn mycontainer.main:app --workers 9 --host 0.0.0.0 --port 8000
